### PR TITLE
Create multiple pd statefulsets

### PR DESCRIPTION
**Pull request by weekface**
Based on the [multiple statefulsets proposal](https://github.com/pingcap/tidb-operator/blob/master/docs/design-proposals/multiple-statefulsets.md), with this pr, we can create multiple pd statefulsets, but can't upgrade or scale them now. Other functions work as usual.

I will open another pr to achieve it.

### CODE REVIEW

File: charts/tidb-cluster/templates/tidb-cluster.yaml [hunk 1]
```diff
@@ -33,6 +33,28 @@ spec:
...
+    {{- if .resources }}
+{{ toYaml .resources | indent 6 }}
+    {{- end }}
+    {{- if .nodeSelector }}
+      nodeSelector:
+{{ toYaml .nodeSelector | indent 8 }}
+    {{- end }}
+      nodeSelectorRequired: {{ .nodeSelectorRequired | default true }}
+    {{- if .tolerations }}
+      tolerations:
```
Comment (onlymellb) [inline charts/tidb-cluster/templates/tidb-cluster.yaml]: ditto

**Comment by weekface** [comment]
Unlike `nodeSelector`, the `resources` attribute doesn't exist in the `TidbCluster` object. Since it was different before, it will cause incompatibility if it is modified now. @only

File: pkg/controller/controller_utils.go [hunk 2]
```diff
@@ -130,8 +130,12 @@ func GetPushgatewayImage(cluster *v1alpha1.TidbCluster) string {
 }
 
 // PDMemberName returns pd member name
-func PDMemberName(clusterName string) string {
-	return fmt.Sprintf("%s-pd", clusterName)
+func PDMemberName(clusterName string, setName ...string) string {
+	if len(setName) == 0 || setName[0] == "" {
```
Comment (onlymellb) [inline pkg/controller/controller_utils.go]: What would cause the slice `setName ` to be non-empty but the first element is empty?

Comment (weekface) [reply pkg/controller/controller_utils.go]: The [main ](https://github.com/pingcap/tidb-operator/pull/265/files#diff-d70b6a464792e09ec86d28a6584a9249R89) PD's [name](https://github.com/pingcap/tidb-operator/pull/265/files#diff-d70b6a464792e09ec86d28a6584a9249R110) is `""`

File: pkg/discovery/discovery.go [hunk 3]
```diff
@@ -81,6 +81,9 @@ func (td *tidbDiscovery) Discover(advertisePeerUrl string) (string, error) {
 	keyName := fmt.Sprintf("%s/%s", ns, tcName)
 	// TODO: the replicas should be the total replicas of pd sets.
```
Comment (onlymellb) [inline pkg/discovery/discovery.go]: Delete this commet？

File: pkg/manager/member/pd_member_manager.go [hunk 4]
```diff
@@ -168,25 +168,21 @@ func (pmm *pdMemberManager) syncPDStatefulSetForTidbCluster(tc *v1alpha1.TidbClu
...
-		err = SetLastAppliedConfigAnnotation(newPDSet)
-		if err != nil {
-			return err
-		}
-		if err := pmm.setControl.CreateStatefulSet(tc, newPDSet); err != nil {
-			return err
-		}
-		tc.Status.PD.StatefulSet = &apps.StatefulSetStatus{}
-		return controller.RequeueErrorf("TidbCluster: [%s/%s], waiting for PD cluster running", ns, tcName)
+	if oldPDSet, err = pmm.setLister.StatefulSets(ns).Get(controller.PDMemberName(tcName, tc.Spec.PD.Name)); err != nil && !errors.IsNotFound(err) {
```
Comment (tennix) [inline pkg/manager/member/pd_member_manager.go]: What if err is NotFound?

**Comment by gregwebs** [comment]
Will this PR be merged or closed?

**Comment by weekface** [comment]
Re-open it if needed.

**Pull request closed without merging**